components/common/hooks/useOpenPricingModal.ts

```diff
@@ -1,17 +1,22 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {useDispatch} from 'react-redux';
+import {useDispatch, useSelector} from 'react-redux';
 
 import {trackEvent} from 'actions/telemetry_actions';
 import {openModal} from 'actions/views/modals';
 import {ModalIdentifiers} from 'utils/constants';
 import PricingModal from 'components/pricing_modal';
 
-export default function useOpenPricingModal() {
+import {isCurrentLicenseCloud} from 'mattermost-redux/selectors/entities/cloud';
+
+export default function useOpenPricingModal(callerComponent?: string) {
     const dispatch = useDispatch();
+    const isCloud = useSelector(isCurrentLicenseCloud);
     return () => {
-        trackEvent('cloud_pricing', 'click_open_pricing_modal');
+        if (isCloud) {
+            trackEvent('cloud_pricing', 'click_open_pricing_modal', {callerComponent});
```

Comment:
Do we need to track the feature flag variant here? On the data side, we know what variants a server has from the rest of the config so this can be mapped when analizing the test results